src/middleware/default_headers.rs

```diff
@@ -0,0 +1,70 @@
+use futures::future::FutureObj;
+
+use http::{
+    header::{HeaderValue, IntoHeaderName},
+    HeaderMap, HttpTryFrom,
+};
+
+use crate::{head::Head, Middleware, Response};
+
+#[derive(Clone)]
+pub struct DefaultHeaders {
+    headers: HeaderMap,
+}
+
+// type KeyValues = [&'static str; 2];
+
+impl Default for DefaultHeaders {
+    fn default() -> Self {
+        DefaultHeaders {
+            headers: HeaderMap::new(),
+        }
+    }
+}
+
+impl DefaultHeaders {
+    pub fn new() -> DefaultHeaders {
+        DefaultHeaders::default()
+    }
+
+    #[inline]
+    #[cfg_attr(feature = "cargo-clippy", allow(clippy::match_wild_err_arm))]
+    pub fn header<K, V>(mut self, key: K, value: V) -> Self
+    where
+        K: IntoHeaderName,
+        HeaderValue: HttpTryFrom<V>,
+    {
+        match HeaderValue::try_from(value) {
+            Ok(value) => {
+                self.headers.append(key, value);
+            }
+            Err(_) => panic!("Cannot create header key value pair"),
```

Comment:
I'm usually with ya here, I think this should be caught while users are developing though? I think crashing would be the right move here but I could be wrong :P Does that make sense to you?

Reply:
Okay, that makes sense; my concern was, what if this is fed with untrusted user input? But this middleware is for default header values (which is likely static or from config files), so panicking on failure might be ok.

Reply:
Yeah I would imagine this would be from a config. It's possible it might come from an untrusted source but I think even then, failing and getting an alert would be preferable to silently failing and not knowing until later, but that could just be me :)